src/Randomizer.SMZ3/Generation/GameHintService.cs

```diff
@@ -333,7 +333,9 @@ private string GetLocationsName(World hintPlayerWorld, IEnumerable<Location> loc
             }
             else if (locations.All(x => x.Room != null))
```

Comment:
I think with these changes, this section should probably be changed. If you take a look at where GetLocationHints is called, multiple are sent in a series of locations. This is just checking if they all are in a room, and while it worked before (though admittedly my logic here was flawed to begin with), I think it'll cause problems now.

What this if statement should do I think is check to see if all of the locations have the _same_ room, not just that they have a room.

Reply:
Good catch. This changed the "Upper Norfair - Crocomire" hint into a hint about "Crocomire's Room" only. Should be fixed now.